sound/soc/sof/topology.c

```diff
@@ -2748,15 +2784,6 @@ void snd_sof_free_topology(struct snd_sof_dev *sdev)
 		return;
 	}
 
-	/* remove routes */
-	list_for_each_entry_safe(sroute, temp, &sdev->route_list, list) {
-		sof_route_remove(&sroute->route);
-
-		/* free sroute and its private data */
-		kfree(sroute->private);
-		kfree(sroute);
-	}
-
 	ret = snd_soc_tplg_component_remove(sdev->component,
 					    SND_SOC_TPLG_INDEX_ALL);
```

Comment:
Do we need snd_soc_tplg_component_remove()?. It is already called by [soc_cleanup_card_resources](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/soc-core.c#L2200) and [__snd_soc_unregister_component](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/soc-core.c#L3281)

Reply:
@bardliao as we discussed, I need to keep the call to snd_soc_tplg_component_remove() to guarantee that sdev->tplg_loaded is set to false only after this call.